core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/UsageUtils.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import com.microsoft.azure.synapse.ml.logging.Usage.HostEndpointUtils._
+import com.microsoft.azure.synapse.ml.logging.Usage.TokenUtils.getAccessToken
+import com.microsoft.azure.synapse.ml.logging.common.SparkHadoopUtils
+
+import java.time.Instant
+import java.util.UUID
+import org.apache.spark.sql.SparkSession
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+
+object UsageTelemetry extends FabricConstants with SparkHadoopUtils with WebUtils {
+  private val SC = SparkSession.builder().getOrCreate().sparkContext
+  private val CapacityId = getHadoopConfig("trident.capacity.id", Some(SC))
+  private val WorkspaceId: String = getHadoopConfig("trident.artifact.workspace.id", Some(SC))
+  private val PbiEnv = SC.getConf.get("spark.trident.pbienv", "").toLowerCase()
+
+  private val SharedHost = getMlflowSharedHost(PbiEnv)
+  private val SharedEndpoint = f"{SharedHost}/metadata/workspaces/{WorkspaceId}/artifacts"
+  private val WlHost = getMlflowWorkloadHost(PbiEnv, CapacityId, WorkspaceId, Some(SharedHost))
```

Comment:
make these lazy so that they dont trigger instantiation of spark. Ideally just put all this stuff inside getMlflowWorkloadHost and make it not take in as many arguments